frontend/src/router/index.ts

```diff
@@ -51,48 +62,85 @@ const router = createRouter({
             name: "project",
             component: () => import("../views/ProjectView.vue"),
             props: (route) => ({ projectId: Number(route.params.projectId) }),
+            meta: {
+                middleware: useCanVisit(
+                    useOrCondition(
+                        useIsStudentOfProjectCondition,
+                        useIsInstructorOfProjectCondition
+                    )
+                ),
+            },
         },
         {
             path: "/project/:projectId(\\d+)/submit",
             name: "onSubmit",
             component: () => import("../views/SubmitView.vue"),
             props: (route) => ({ projectId: Number(route.params.projectId) }),
+            meta: {
+                middleware: useCanVisit(useIsInGroupOfProjectCondition),
+            },
         },
         {
             path: "/project/:projectId(\\d+)/groups",
             name: "groups",
             component: () => import("../views/GroupsView.vue"),
             props: (route) => ({ projectId: Number(route.params.projectId) }),
+            meta: {
+                middleware: useCanVisit(
+                    useOrCondition(
+                        useIsStudentOfProjectCondition,
+                        useIsInstructorOfProjectCondition
+                    )
+                ),
+            },
         },
         {
             path: "/groups/:groupId(\\d+)",
             name: "group",
             component: () => import("../views/GroupView.vue"),
             props: (route) => ({ groupId: Number(route.params.groupId) }),
+            meta: {
+                middleware: useCanVisit(
+                    useOrCondition(useIsInGroupCondition, useIsInstructorOfGroupCondition)
+                ),
+            },
         },
         {
             path: "/project/:projectId(\\d+)/submissions",
             name: "projectSubmissions",
             component: () => import("../views/SubmissionsTeacherView.vue"),
             props: (route) => ({ projectId: Number(route.params.projectId) }),
+            meta: {
+                middleware: useCanVisit(useIsInstructorOfProjectCondition),
+            },
         },
         {
             path: "/subjects",
             name: "subjects",
             component: () => import("../views/subject/SubjectsView.vue"),
-            children: [],
         },
         {
             path: "/subjects/:subjectId(\\d+)",
             name: "subject",
             component: () => import("../views/subject/SubjectView.vue"),
             props: (route) => ({ subjectId: Number(route.params.subjectId) }),
+            meta: {
+                middleware: useCanVisit(
+                    useOrCondition(
+                        useIsStudentOfSubjectCondition,
+                        useIsInstructorOfSubjectCondition
```

Comment:
Als is teacher of admin ben kan ik er niet aan

Reply:
idem opmerking ivm submissions